File: src/cli/config/reset.ts
import type { Command } from "commander";
import { printJson, printError } from "../../lib/output.ts";
import { getSettings, updateSettings, resetSettings } from "../../lib/config.ts";
import type { Settings } from "../../lib/config.ts";
import { SETTING_DEFS, VALID_KEYS } from "./index.ts";

export function registerReset(config: Command): void {
  config
    .command("reset")
    .argument("[key]", "Setting key (omit to reset all)")
    .description("Reset settings to defaults")
    .action((key?: string) => {
      if (!key) {
        resetSettings();
        printJson({ reset: "all" });
        return;
      }

      const def = SETTING_DEFS[key];
      if (!def) {
        printError(`Unknown setting: ${key}. Valid keys: ${VALID_KEYS.join(", ")}`);
        return;
      }

      const [section, field] = def.path;
      const settings = getSettings();
      const sectionObj = settings[section] as Record<string, unknown> | undefined;
      if (sectionObj) {
        delete sectionObj[field];
        updateSettings({ [section]: sectionObj } as Settings);
      }
      printJson({ reset: key });
    });
}

File: internal/cli/configcmd/reset.go
package configcmd

import (
	"fmt"

	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/config"
	"github.com/shhac/lin/internal/output"
)

func registerReset(cfg *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "reset [key]",
		Short: "Reset settings to defaults",
		Args:  cobra.MaximumNArgs(1),
		Run: func(_ *cobra.Command, args []string) {
			if len(args) == 0 {
				if err := config.ResetSettings(); err != nil {
					output.PrintError(err.Error())
					return
				}
				output.PrintJSON(map[string]any{"reset": "all"})
				return
			}

			key := args[0]
			def, ok := settingDefs[key]
			if !ok {
				output.PrintError(fmt.Sprintf("Unknown setting: %s. Valid keys: %s", key, validKeysStr()))
				return
			}

			settings := config.GetSettings()
			def.reset(settings)

			if err := config.UpdateSettings(settings); err != nil {
				output.PrintError(err.Error())
				return
			}
			output.PrintJSON(map[string]any{"reset": key})
		},
	}
	cfg.AddCommand(cmd)
}
